compiler/ir/serialization.js/src/org/jetbrains/kotlin/ir/backend/js/JsGeneratorExtensions.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.ir.backend.js
+
+import org.jetbrains.kotlin.descriptors.PropertyDescriptor
+import org.jetbrains.kotlin.descriptors.Visibility
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi2ir.generators.GeneratorExtensions
+
+class JsGeneratorExtensions : GeneratorExtensions() {
+    override fun computeFieldVisibility(descriptor: PropertyDescriptor): Visibility =
```

Comment:
@skuzmich Please take a look

Reply:
JS IR backend can export properties but not fields. All fields should be private.

Reply:
That's refreshingly simple, thanks! But can you take a look at `testEval` and `testTopLevelCallables`? Both access a top level exported field through reflection and fail if this field is not public. As a compromise, the code currently still marks `@JsExport`ed fields as public. Is it necessary to support access to fields through reflection like this? If not, marking every field as private would work just fine.

Reply:
Tests fail due to bug in JS code generator. They supposed to access top level properties but instead, by mistake, fields are exported with the same name. I'll change `JsGeneratorExtensions#computeFieldVisibility` to return `Visibilities.PRIVATE` once its fixed. Thanks!